tests/js/comments/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -3621,6 +3628,15 @@ function binaryInBinaryLeft() {
   )
 }
 
+function binaryInBinaryLeftDivision() {
+  return (
+    // Reason for 42
+    42 /
+      84 +
+    2
+  )
+}
```

Comment:
It is really ugly, we should fix it here

Reply:
In that case we can open another issue, fix this first

**Prettier 2.1.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEqEEAFAJT7AA6U++ATnDOvbSTXQPSf4BKcAQwDOBIvXwAWAEwdJMqBQCk+ABz4A1PikBuGgF8QAGhAQADrmhDkoAfXoQA7gAVbCKygEAbBwICeV4wAjegEwAGsmAGVTUJwoAHNkGHp0OGM4AFtAuAATHNyAGQEE9AF4uAAxCHoMgRhcBOQQAXQYCCMQAAsYDM8AdU6ceCEYsDhItyGcADch3yawIQCQOKE4ehgnEPja5FQvNeMAKyEADwAhEPCogQy4Ari4PYO0kBPTyLj4zzgARXQIPBnp5DiAYvQ1vQmoEBNlPB1TPQ4jA+jgcjBOsgVAAGYyIiBrPohUxNRFwSHTJ7GACOAPgWzM7maQgAtFA4Llch1GLScIwtmVdkh9iDXmsMjgkikxV8fv9AU9hS9jDBYaj0ZikFIVSEcJ4vgBhCAZIUgckAVg66DWABVYe4RaDpqkAJJQfKwSJgJHmACC7siMF8P2Baz0eiAA)
```sh
--parser babel
```

**Input:**
```jsx
function foo() {
  return (
  // Reason for 42
  42
) % 8 + 2;
}
```

**Output:**
```jsx
function foo() {
  return (
    (// Reason for 42
    42 %
      8) +
    2
  );
}

```

**Second Output:**
```jsx
function foo() {
  return (
    // Reason for 42
    (42 % 8) +
    2
  );
}

```

Reply:
I don't think we should merge this commits with a lot of future issue, instead we should solve it here

Reply:
I mean fix `%` one first, then  this.

Reply:
What would the desired output look like? I tried to walk through the codebase to find the proper place to move/format lines/parens/comments and was kind of lost. I am committed to learning how to work with the prettier codebase to become an active contributor, so I'm going to give it another go. If you wouldn't mind giving me a target to shoot for in the output? @fisker @evilebottnawi

Reply:
This one may won't be easy to fix. Format original one 3 times, you'll get this.


```
function foo() {
  return (
    // Reason for 42
    (42 % 8) + 2
  );
}

```

Reply:
@fisker yeah I had the same result, I tried looking in the parser where needsParens is called and manipulating the insertion point of the parenthesis to be after the comment, but the comment is already printed at that point. I tried copying the pattern that was used for flow type annotations where the comment object is mutated to have `printed = true` and then the comment is manually inserted. I thought that would do the trick but like I said it seems in that logic tree, or with leading comments, it is already printed at the point it gets to inserting the parens?

Reply:
in the example here, if you move the comment in the input to be above the return statement, the output is as you'd want... This is why I was hoping to manipulate the printing of the comment to be before the inserted opening parens.

Reply:
This is really a separate issue, namely https://github.com/prettier/prettier/issues/7746
I don't think it should be fixed here.